src/OrbitGl/CaptureWindow.cpp

```diff
@@ -113,8 +113,21 @@ void CaptureWindow::PreRender() {
   }
 
   if (time_graph_ != nullptr) {
-    time_graph_->UpdateLayout();
-    UpdateChildrenPosAndSize();
+    const int kMaxLayoutLoops =
+        app_ != nullptr && (app_->IsCapturing() || app_->IsLoadingCapture()) ? 1 : 10;
+
+    int layout_loops = 0;
+
+    // Layout changes of one element may require other elements to be updated as well,
+    // so layouting needs to be done until all elements report that they do not need to
+    // be updated further. As layout requests bubble up, it's enough to check this for
+    // the root element (time graph) of the tree.
+    // During loading or capturing, only a single layouting loop is executed as we're
+    // streaming in data from a seperate thread.
+    do {
+      UpdateChildrenPosAndSize();
+      time_graph_->UpdateLayout();
```

Comment:
Shouldn't first the timegraph update its layout and after UpdateChildrenPosAndSize()?

Reply:
It depends what's happening in both calls, and due to the loop, it doesn't matter. Either way can cause issues: Time graph layouting could influence the children (e.g. it could update something related to visible time), but child layouting could also influence the time graph (e.g. if the scrollbar will be updated during TimeGraph::UpdateLayout). The goal is that this is robust in all cases.

So I decided on the following convention: `UpdateLayout` should be the last thing that happens before we start drawing. I'll write that in a documentation about the whole update / render process.

Reply:
Sounds good! :)